src/mesh/p4est_mesh.jl

```diff
@@ -0,0 +1,273 @@
+"""
+    P4estMesh{NDIMS, RealT<:Real} <: AbstractMesh{NDIMS}
```

Comment:
Do you really want to expose `RealT` as part of the API?

Reply:
I don't know, it's like this in `CurvedMesh` too. Should we remove this from the docs?

Reply:
I would prefer doing that. It's not a hard requirement, but it's easier for us if we don't expose that part officially.